pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -184,6 +183,13 @@ func buildClusterHosts(env *model.Environment, service *model.Service, port int)
 
 	hosts := make([]*core.Address, 0)
 	for _, instance := range instances {
+		// Only send endpoints from the networks in the network view requested by the proxy.
+		// The default network view assigned to the Proxy is the UnnamedNetwork (""), which matches
+		// the default network assigned to endpoints that don't have an explicit network
+		if !proxyNetworkView[instance.Endpoint.Network] {
```

Comment:
I assume this is based on golang returning 'false' if entry not found ...

Reply:
It is, I asked him to change from the clunkier `if _, found := proxyNetworkView[instance.Endpoint.Network]; !found` since this is a `map[string]bool`. We use the reverse: `if proxyNetworkView[instance.Endpoint.Network]` all the time.